libs/blocks/icon-block/icon-block.js

```diff
@@ -54,13 +54,31 @@ function decorateContent(el) {
   if (text) {
     text.classList.add('text-content');
     const image = block.querySelector(':scope img');
-    if (image) image.closest('p').classList.add('icon-area');
-    // place standalone links inside an action-area
     const lastElem = text.lastElementChild;
+    let actionLink = null;
     if (lastElem.children.length === 1
       && lastElem.lastElementChild.nodeName === 'A'
       && lastElem.lastElementChild.innerText === lastElem.innerText) {
       text.lastElementChild.classList.add('action-area');
+      actionLink = lastElem.lastElementChild;
+    }
+
+    if (image) {
+      const iconP = image.closest('p');
+      iconP.classList.add('icon-area');
+      const iconLink = iconP.querySelector('a');
```

Comment:
```suggestion
      iconP?.classList.add('icon-area');
      const iconLink = iconP?.querySelector('a');
```
In certain scenarios, the markup could be different, better safe than sorry